### PR TITLE
Improve graph node labels and PF badging

### PR DESCRIPTION
**Pull request by jshaughn**
Fixes https://github.com/kiali/kiali/issues/3879

Improve graph badging, ensure namespace and/or cluster information is provided where needed for discrimination, and not provided when it's not.
- remove boxByCluster and boxByNamespace booleans from cy's global scratch data, no longer needed
- add homeCluster to Cy's global scratch data
- beef up side panel badge tooltips with cluster+namespace discriminators, this avoids having to display a lot more badges/values in the side panel
- This PR includes and associated refactor that touches files that apply PF badges:
  - Consolidate PF Badging in a way sort of analogous to PfColors...
  - Basically put all all of our used PF Badges in one place for easy reference, add some stronger typing, and provide some utilities for consistent use and styling.  Update current PF badges to use the new approach.
  - This refactor should not affect the styling of existing PF badges, but it makes it easier to ensure consistent styling going forward, and to easy see our currently defined badges and their associated abbreviations and default tooltips.

### CODE REVIEW

**Comment by jshaughn** [comment]
@israel-hdez Adding you as reviewer for code and to ensure all labels look to have the correct info in MC scenarios.  Also, validate the enhanced badge tooltips in the side panel.
@lucasponce Adding you as reviewer mainly for the new PfBadges refactor because it touches files across the UI.

File: src/pages/WorkloadDetails/WorkloadPodLogs.tsx [hunk 1]
```diff
@@ -257,9 +257,7 @@ class WorkloadPodLogs extends React.Component<WorkloadPodLogsProps, WorkloadPodL
                     {this.state.showToolbar && (
                       <Toolbar className={toolbar}>
                         <ToolbarGroup>
-                          <Tooltip position={TooltipPosition.top} content={<>Pod</>}>
-                            <Badge className="virtualitem_badge_definition">P</Badge>
-                          </Tooltip>
+                          {pfBadge(PFBadges.Pod)}
```
Comment (israel-hdez) [inline src/pages/WorkloadDetails/WorkloadPodLogs.tsx:260]: position missing?

File: src/components/VirtualList/Renderers.tsx [hunk 2]
```diff
@@ -189,9 +168,7 @@ export const namespace: Renderer<TResource> = (item: TResource) => {
       key={'VirtuaItem_Namespace_' + item.namespace + '_' + item.name}
       style={{ verticalAlign: 'middle' }}
     >
-      <Tooltip position={TooltipPosition.top} content={<>Namespace</>}>
-        <Badge className={'virtualitem_badge_definition'}>NS</Badge>
-      </Tooltip>
+      {pfBadge(PFBadges.Namespace)}
```
Comment (israel-hdez) [inline src/components/VirtualList/Renderers.tsx:171]: position missing?

**Review by israel-hdez — COMMENTED** [review]
I see it works.
I get the (namespace) and the (cluster) legends in graph labels as needed. It provides much, much better context.
I also can see the new "Cluster: xyz" text when hovering on the sidepanel badges.

For the functionality, I'm quite good with this (although, I didn't try all affected pages).

About the code, I have some minor comments.
I miss using react syntax rather than calling a function. But it's no problem, you can keep as is.

File: src/components/Link/IstioObjectLink.tsx [hunk 3]
```diff
@@ -32,9 +32,7 @@ export class ReferenceIstioObjectLink extends React.Component<Props> {
 
     return (
       <>
-        <Tooltip position={TooltipPosition.top} content={<>{istioType.name}</>}>
-          <Badge className={'virtualitem_badge_definition'}>{istioType.icon}</Badge>
-        </Tooltip>
+        {pfBadge(istioType.badge)}
```
[the remark this answers is not in the document]
Comment (jshaughn) [reply src/components/Link/IstioObjectLink.tsx:35]: good catch. I'll fix.

File: src/components/IstioWizards/RequestRouting/Rules.tsx [hunk 4]
```diff
@@ -137,9 +138,7 @@ class Rules extends React.Component<Props> {
                       .map((wk, i) => {
                         return (
                           <div key={'wk_' + order + '_' + wk.name + '_' + i}>
-                            <Tooltip position={TooltipPosition.top} content={<>Workload</>}>
-                              <Badge className={'virtualitem_badge_definition'}>WS</Badge>
-                            </Tooltip>
+                            {pfBadge(PFBadges.Workload, TooltipPosition.top)}
```
[the remark this answers is not in the document]
Comment (jshaughn) [reply src/components/IstioWizards/RequestRouting/Rules.tsx:141]: I don't have a great answer.  I think it just sort of came out this way.  But, yeah, maybe it would be more natural.  I'll sleep on it and let you know tomorrow,

File: src/components/VirtualList/Renderers.tsx (hunk 2 above)
Comment (jshaughn) [reply src/components/VirtualList/Renderers.tsx:171]: I must have been tired.  Good eye.

File: src/components/IstioWizards/RequestRouting/Rules.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jshaughn) [reply src/components/IstioWizards/RequestRouting/Rules.tsx:141]: I didn't sleep, I changed it, please take a look at the update.

**Comment by lucasponce** [comment]
Note I'm hitting this https://github.com/kiali/kiali/issues/3911 while testing it in my local enviornment.

**Review by israel-hdez — APPROVED** [review]
Code is more clear!
I retested an all looks good.

Also works fine when merging #2129

**Comment by jshaughn** [comment]
@lucasponce I have rebased the PR on master and so you should no longer hit the issue that prevented the review.  Please take another look, thanks.

**Review by lucasponce — APPROVED** [review]
LGTM thanks !

I've fetched locally and tested features where badges are used and it seems good.

**Comment by pbajjuri20** [comment]
Tested and passed on Kiali UI 1.34.0 (0c141c6a253a64808211239881dfd43d7f96b5f8)

**Review by pbajjuri20 — APPROVED** [review]
